types/chrome/index.d.ts

```diff
@@ -5998,6 +5998,63 @@ declare namespace chrome.notifications {
     export function getPermissionLevel(callback: (level: string) => void): void;
 }
 
+////////////////////
+// Offscreen
+////////////////////
+/**
+ * Use the offscreen API to create and manage offscreen documents.
+ * Availability: Since Chrome 109. Manifest v3.
+ * Permissions: "offscreen"
+ */
+declare namespace chrome.offscreen {
+    /** The reason(s) the extension is creating the offscreen document. */
+    export enum Reason {
+        /** A reason used for testing purposes only. */
+        TESTING,
+        /** The offscreen document is responsible for playing audio. */
+        AUDIO_PLAYBACK,
+        /** The offscreen document needs to embed and script an iframe in order to modify the iframe's content. */
+        IFRAME_SCRIPTING,
+        /** The offscreen document needs to embed an iframe and scrape its DOM to extract information. */
+        DOM_SCRAPING,
+        /** The offscreen document needs to interact with Blob objects (including URL.createObjectURL()). */
+        BLOBS,
+        /** The offscreen document needs to use the DOMParser API. */
+        DOM_PARSER,
+        /** The offscreen document needs to interact with media streams from user media (e.g. getUserMedia()). */
+        USER_MEDIA,
+        /** The offscreen document needs to interact with media streams from display media (e.g. getDisplayMedia()). */
+        DISPLAY_MEDIA,
+        /** The offscreen document needs to use WebRTC APIs. */
+        WEB_RTC,
+        /** The offscreen document needs to interact with the clipboard APIs(e.g. Navigator.clipboard). */
+        CLIPBOARD
+    }
+
+    /** The parameters describing the offscreen document to create. */
+    export interface CreateParameters {
+        /** The reason(s) the extension is creating the offscreen document. */
+        reasons: Reason[];
+        /** The (relative) URL to load in the document. */
+        url: string;
+        /** A developer-provided string that explains, in more detail, the need for the background context. The user agent _may_ use this in display to the user. */
+        justification: string;
+    }
+
+    /**
+     * Creates a new offscreen document for the extension.
+     * @param parameters The parameters describing the offscreen document to create.
+     * @param callback Invoked when the offscreen document is created and has completed its initial page load.
+     */
+    export function createDocument(parameters: CreateParameters, callback: () => void): void;
+    /**
+     * Closes the currently-open offscreen document for the extension.
+     * @param callback Invoked when the offscreen document has been closed.
+     */
+    export function closeDocument(callback: () => void): void;
```

Comment:
Can you change this to return `Promise<void>;` so it matches other parts of this file?

Reply:
In the [documentation](https://developer.chrome.com/docs/extensions/reference/offscreen/), it's saying that, by not specifying the callback, the return will be async. However, as this overloading method is listed as _Pending_, I didn't add it.
Would you like that I list these two overloading options or just change the return to a promise?

Reply:
Thanks for clarifying for me, I missed the pending note. I'd leave it as-is for now.